### PR TITLE
- adapt the client for preselecting pattern to work also in normal stage

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

**Comment by jsuchome** [comment]
See also https://build.suse.de/request/show/82661

File: src/clients/inst_crowbar_patterns.rb [hunk 1]
```diff
@@ -28,12 +28,47 @@ class InstCrowbarPatternsClient < Client
...
+        Pkg.SourceStartManager(true)
+        patterns.each do |pattern|
+          Pkg.ResolvableInstall(pattern, :pattern)
+        end
+
+        ret = PackagesUI.RunPatternSelector
+        Builtins.y2milestone("RunPatternSelector returned %1", ret)
+
+        dialog_ret = :next
+        dialog_ret = :abort if ret == :cancel
```
Comment (lslezak) [inline src/clients/inst_crowbar_patterns.rb]: NP: I'd join this to a single line: `dialog_ret =  (ret == :cancel) ? :abort : :next`

File: src/clients/inst_crowbar_patterns.rb [hunk 2]
```diff
@@ -28,12 +28,47 @@ class InstCrowbarPatternsClient < Client
...
+        # inst_add-on_software to setup source and target
+
+        Pkg.TargetInit(Installation.destdir, false)
+        Pkg.SourceStartManager(true)
+        patterns.each do |pattern|
+          Pkg.ResolvableInstall(pattern, :pattern)
+        end
+
+        ret = PackagesUI.RunPatternSelector
+        Builtins.y2milestone("RunPatternSelector returned %1", ret)
```
Comment (lslezak) [inline src/clients/inst_crowbar_patterns.rb]: Use the new logging functions in the new code, `Builtins.*` are obsolete. See http://www.rubydoc.info/github/yast/yast-ruby-bindings/master/Yast/Logger.

Comment (jsuchome) [inline src/clients/inst_crowbar_patterns.rb]: Cool, it was about time! :-)

**Comment by lslezak** [comment]
Just minor issues, LGTM in general.

File: src/clients/inst_crowbar_patterns.rb [hunk 3]
```diff
@@ -25,16 +25,51 @@
...
+
+        ret = PackagesUI.RunPatternSelector
+        log.info("RunPatternSelector returned #{ret}")
+
+        dialog_ret = :next
+        dialog_ret = :abort if ret == :cancel
+
+        if ret == :accept || ret == :ok
+          # Add-on requires packages to be installed right now
+          log.info("Selected resolvables will be installed now")
```
Comment (lslezak) [inline src/clients/inst_crowbar_patterns.rb]: <sub>Really, really small nit picking :wink: : the convention is to not use brackets for the logging calls (like in the linked documentation), it's similar to `puts "foo"`.</sub>

**Comment by lslezak** [comment]
Great, thanks! :+1: LGTM

**Pull request merged**